### PR TITLE
Ensure gFTL is installed with Pyccel

### LINKED ISSUE #1656: Ensure gFTL is installed with Pyccel
### Relevant Discussion

https://github.com/pyccel/pyccel/discussions/1053#discussioncomment-6891446

### Describe the feature
In order to add support for containers we need to include our new dependencies in Pyccel and ensure that they are correctly installed. In C we will use https://github.com/Goddard-Fortran-Ecosystem/gFTL . We need to make sure that the necessary files for compilation are packaged with Pyccel.

### Proposed Solution
Add https://github.com/Goddard-Fortran-Ecosystem/gFTL as a git submodule. Update the `pyproject.toml` file to ensure that the necessary files from STC are installed with Pyccel. We do not need to install the docs from the library.

There may need to be an investigation to find the most appropriate way to handle this. I think it would be a good idea to create a folder called `extensions` at the root and keep dependencies here. It is not clear how to map the include and src folders to the Pyccel folder from there. The solution will involve artifacts and may require symbolic links

### PR DESCRIPTION
**Pull request by mustapha-belbiad**
- Add  `https://github.com/Goddard-Fortran-Ecosystem/gFTL` as a git submodule.
- Update the `pyproject.toml` file to include essential files from gFTL.
- Implement a GitHub Action in `check_for_gftl` to verify gFTL installation.

This fixes #1656.

### CODE REVIEW

File: .github/workflows/pickle_wheel.yml [hunk 1]
```diff
@@ -82,8 +82,16 @@ jobs:
...
+      - name: Check for gFTL installation
+        id: gFTL_check
+        uses: ./.github/actions/check_for_gftl
       - name: "Post completed"
         if: always()
         run: |
-          python ci_tools/basic_json_check_output.py --statuses ${{ steps.pickle.outcome }} ${{ steps.pickle_check.outcome }} ${{ steps.stc_check.outcome }} --reasons "Installation failed." "Pickled files were not found in installaion." "STC was not found during installation."
-          python ci_tools/complete_check_run.py ${{ steps.pickle.outcome }} ${{ steps.pickle_check.outcome }} ${{ steps.stc_check.outcome }}
+          python ci_tools/basic_json_check_output.py --statuses ${{ steps.pickle.outcome }} ${{ steps.pickle_check.outcome }} ${{ steps.gFTL_check.outcome}} --reasons "Installation failed." "Pickled files were not found in installaion." "gFTL was not found during installation."
+          python ci_tools/complete_check_run.py ${{ steps.pickle.outcome }} ${{ steps.pickle_check.outcome }} ${{ steps.gFTL_check.outcome}}
```
Comment (EmilyBourne) [inline .github/workflows/pickle_wheel.yml:92]: You seem to have lost the stc check here

File: pyproject.toml [hunk 2]
```diff
@@ -49,19 +49,21 @@ artifacts = [
...
   "pyccel/stdlib/*/*.f90",
-  "pyccel/extensions/STC/include/*/*.c"
+  "pyccel/extensions/STC/include/*/*.c",
+  "pyccel/extensions/gFTL/include/v2"
 ]
 exclude = [
     "pyccel/extensions/STC/src",
-    "pyccel/extensions/STC/misc"
+    "pyccel/extensions/STC/misc",
+    "pyccel/extensions/gFTL/tests"
```
Comment (EmilyBourne) [inline pyproject.toml:66]: Do you know which `include` is causing the `tests/` folder to be included? I don't see any Python files

Comment (mustapha-belbiad) [reply pyproject.toml:66]: I will remove this line in my next commit. 
The reason for excluding the `tests` folder is that it was added with `include`, but now that I have changed it to 'pyccel/extensions/STC/include', there is no need to exclude it.

File: .github/workflows/pickle_wheel.yml (hunk 1 above)
Comment (mustapha-belbiad) [reply .github/workflows/pickle_wheel.yml]: Fixed : 4be4d3c

File: pyproject.toml (hunk 2 above)
Comment (mustapha-belbiad) [reply pyproject.toml]: Fixed : a06f44e

File: pyproject.toml [hunk 3]
```diff
@@ -49,15 +49,16 @@ artifacts = [
...
   "*.py",
   "*.pyh",
   "*.pyccel",
   "pyccel/parser/grammar/*.tx",
   "pyccel/stdlib/*/*.h",
   "pyccel/stdlib/*/*.c",
   "pyccel/stdlib/*/*.f90",
-  "pyccel/extensions/STC/include/*/*.c"
+  "pyccel/extensions/STC/include/*/*.c",
+  "pyccel/extensions/gFTL/include/v2"
```
Comment (EmilyBourne) [inline pyproject.toml:61]: Can we reorder this to group the extensions together please

Comment (mustapha-belbiad) [reply pyproject.toml]: Done : 2cb3c6e

File: pyproject.toml [hunk 4]
```diff
@@ -49,15 +49,16 @@ artifacts = [
...
   "*.py",
   "*.pyh",
   "*.pyccel",
   "pyccel/parser/grammar/*.tx",
   "pyccel/stdlib/*/*.h",
   "pyccel/stdlib/*/*.c",
   "pyccel/stdlib/*/*.f90",
-  "pyccel/extensions/STC/include/*/*.c"
+  "pyccel/extensions/STC/include",
+  "pyccel/extensions/STC/include/*/*.c",
```
Comment (EmilyBourne) [inline pyproject.toml:60]: Are both of these lines needed?

Comment (mustapha-belbiad) [reply pyproject.toml:60]: I think the first line will add all folders and files, so I will remove the second line.

Comment (mustapha-belbiad) [reply pyproject.toml]: Fixed : db739e5

**Review by EmilyBourne — APPROVED** [review]
Looks good to me

**Review by yguclu — CHANGES_REQUESTED** [review]
Looks good to me.

File: .github/actions/check_for_gftl/action.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    - name: check gFTL installation
+      shell: python
+      run: |
+        import pyccel
+        from os.path import exists, join
+        pyccel_loc = pyccel.__path__[0]
+        print(pyccel_loc)
+        assert(exists(join(pyccel_loc,'extensions/gFTL/include/v2')))
+        if ${{ inputs.not_editable }}:
+          assert(not exists(join(pyccel_loc,'extensions/gFTL/doc')))
```
[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/actions/check_for_gftl/action.yml]: Fixed : 3c4ee70

[the remark this answers is not in the document]
Comment (yguclu) [reply .github/actions/check_for_gftl/action.yml]: Did you read my comment in full?

By the way, you could have just clicked on "Commit suggestion".

[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/actions/check_for_gftl/action.yml]: I think the issue was with my code editor, but it has been resolved now :  0c574f4

[the remark this answers is not in the document]
Comment (yguclu) [reply .github/actions/check_for_gftl/action.yml]: Nope... I also wrote "`assert` is not a function in Python" and removed the parentheses.

[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/actions/check_for_gftl/action.yml]: Sorry, Fixed : b7ef515
Should I do the same for STC in this PR?

[the remark this answers is not in the document]
Comment (yguclu) [reply .github/actions/check_for_gftl/action.yml]: Yes, good idea

[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/actions/check_for_gftl/action.yml]: Fixed : 9859342

**Review by yguclu — APPROVED** [review]

**Pull request merged**